Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -462,9 +462,9 @@ internal bool OnEndRequest_ShouldLog(HttpContext context)
                 try
                 {
                     var rootRequestId = headers[HeaderRootRequestId];
-                    rootRequestId = StringUtilities.EnforceMaxLength(rootRequestId, InjectionGuardConstants.RequestHeaderMaxLength);
                     if (rootRequestId != null)
                     {
+                        rootRequestId = StringUtilities.EnforceMaxLength(rootRequestId, InjectionGuardConstants.RequestHeaderMaxLength);
```

Comment:
same as before:

we should always expect a header value to be null

EnforceMaxLength has Debug.Assert that value is not null preventing to run tests locally in debug.

Instead, I can remove Debug.Assert from the EnforceMaxLength, however I don;t know what was the motivation putting it there in the first place.

@MS-TimothyMothra,
It seems you've done this change. could you please advise what is the best approach here?

Reply:
Please remove the Debug.Assert. (see comment above)